portality/bll/services/concurrency_prevention.py

```diff
@@ -0,0 +1,14 @@
+from portality.core import app
+from portality.bll.exceptions import ConcurrentUpdateRequestException
+from portality.ui.messages import Messages
+import redis
+
+class UpdateRequestConcurrencyPreventionService:
```

Comment:
I think the name of this class has changed since my last review, which makes this specific to Update Requests.

I'm in two minds about this:

In one way, this is good because it makes the class specific to its usage, and we can then make others for `ArticleConcurencyProventionService` or whatever.

On the other hand, this entire service really only has one function, and every time we want to do something like this for another type or in another context, we need another BLL service.

So, I would do this instead in one of two ways:

1. Have a generic concurrency service with generic capabilities
2. Add the concurrency prevention capabilities to the appropriate services for the type.  So, preventing concurrency for update would either go on the `ApplicationService` (since URs are a kind of application), OR we start a new `UpdateRequestsService` and put it there.

Then, later, when we do this for articles, we add the `prevent_concurrency` method to the `ArticleService`.

My feeling is that the former approach (General Concurrency Service) is better, because it needs to manage a Redis connection, and it would be best to have that all in one place.

There is a third path, which is to do both options: have a generic concurrency service, that in turn is wrapped by specific services on the the type-specific services.  That gives you both benefits: a concurrency method that can be specific to the needs, and a centrally managed access to Redis.